src/InstrumentationEngine.Lib/Logging.cpp

```diff
@@ -174,6 +177,15 @@ HRESULT CLogging::SetLogFileLevel(_In_ LoggingFlags fileLogFlags)
     return s_loggerService.Get()->SetLogFileLevel(fileLogFlags);
 }
 
+// static
+HRESULT CLogging::SetCachedLoggingFlag(_In_ LoggingFlags flags)
+{
+    // Do not check initialization as that can cause an infinite loop.
```

Comment:
nit: If you want to keep things as-is and not have to check initialization, you could tighten this up by making this function private, and declaring CLoggerService as a friend class. That way no one else accidentally calls it.

Some people don't like using friend classes, though, so, I'm not tied to the idea.